docs/book/src/tasks/experimental-features/runtime-sdk/implement-topology-mutation-hook.md

```diff
@@ -0,0 +1,182 @@
+# Implementing Topology Mutation Hook Runtime Extensions
+
+<aside class="note warning">
+
+<h1>Caution</h1>
+
+Please note Runtime SDK is an advanced feature. If implemented incorrectly, a failing Runtime Extension can severely impact the Cluster API runtime.
+
+</aside>
+
+## Introduction
+
+The Topology Mutation Hooks are going to be called during each Cluster topology reconciliation. More specifically 
+we are going to call two different hooks for each reconciliation:
+
+* **GeneratePatches**: GeneratePatches is responsible for generating patches for the entire Cluster topology.
+* **ValidateTopology**: ValidateTopology is called after all patches have been applied and thus allow to validate 
+  the resulting objects.
+
+![Cluster topology reconciliation](../../../images/runtime-sdk-topology-mutation.png)
+
+Please see the corresponding [CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20220330-topology-mutation-hook.md)
+for additional background information. 
+
+## Inline vs. external patches
+
+Inline patches have the following advantages:
+* Inline patches are easier when getting started with ClusterClass as they are built into
+  the Cluster API core controller, no external component have to be developed and managed. 
+
+External patches have the following advantages:
+* External patches can be individually written, unit tested and released/versioned.
+* External patches can leverage the full feature set of a programming language and 
+  are thus not limited to the capabilities of JSON patches and Go templating.
+* External patches can use external data (e.g. from cloud APIs) during patch generation.
+* External patches can be easily reused across ClusterClasses.
+
+## Using one or multiple external patch extensions
+
+Some considerations:
+* In general a single external patch extension is simpler than many, as only one extension 
+  then has to be built, deployed and managed. 
+* A single extension also requires less HTTP round-trips between the CAPI controller and the extension(s).
+* With a single extension it is still possible to implement multiple logical features using different variables.
+* When implementing multiple logical features in one extension it's recommended that they can be conditionally
+  enabled/disabled via variables (either via certain values or by their existence).
+* [Conway's law](https://en.wikipedia.org/wiki/Conway%27s_law) might make it not feasible in large organizations 
```

Comment:
😆

Reply:
Fabrizio challenged me if I'm able to place that quote in there :)